### PR TITLE
HDDS-10085. Improve Method Name in ContainerBalancerSelectionCriteria

### PR DESCRIPTION
**Pull request by siddhantsangwan**
## What changes were proposed in this pull request?
Renamed the following method to `isECContainerAndLegacyRMEnabled`. `isECContainer` was misleading, because Container Balancer supports balancing EC containers. This PR only changes the method names and updates java doc. No logic change.

```
  private boolean isECContainer(ContainerInfo container) {
    return container.getReplicationType().equals(HddsProtos.ReplicationType.EC)
        && replicationManager.getConfig().isLegacyEnabled();
  }
```

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10085

## How was this patch tested?

Ran existing tests.

### CODE REVIEW

**Comment by siddhantsangwan** [comment]
@umamaheswararao @adoroszlai @sodonnel Can someone please review? Just a method name and java doc improvement.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerSelectionCriteria.java [hunk 1]
```diff
@@ -158,12 +155,12 @@ private Comparator<ContainerID> orderContainersByUsedBytes() {
...
   /**
    * Checks whether a Container has the ReplicationType
-   * {@link HddsProtos.ReplicationType#EC}.
+   * {@link HddsProtos.ReplicationType#EC} and the Legacy Replication Manger is enabled.
    * @param container container to check
    * @return true if the ReplicationType is EC and "hdds.scm.replication
    * .enable.legacy" is true, else false
    */
-  private boolean isECContainer(ContainerInfo container) {
+  private boolean isECContainerAndLegacyRMEnabled(ContainerInfo container) {
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerSelectionCriteria.java:163]: Maybe it's just me, but this suggests to me as if `Enabled` applies to both legacy RM and EC container.

How about `isUnsupportedType` (or similar), to hide internal details?

**Review by adoroszlai — COMMENTED** [review]
Thanks @siddhantsangwan for the patch, LGTM.

I was looking at it earlier, but didn't want to nitpick on the name.  I'm fine with the existing name, the new one, or the one I suggest. :)

**Review by adoroszlai — APPROVED** [review]

**Comment by siddhantsangwan** [comment]
Thanks @adoroszlai for the review.

**Pull request merged**